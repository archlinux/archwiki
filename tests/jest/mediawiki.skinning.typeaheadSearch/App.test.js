const VueTestUtils = require( '@vue/test-utils' );
const App = require( '../../../resources/src/mediawiki.skinning.typeaheadSearch/App.vue' );
const urlGeneratorFn = require( '../../../resources/src/mediawiki.skinning.typeaheadSearch/urlGenerator.js' );
const scriptPath = '/w/index.php';
const urlGenerator = urlGeneratorFn( scriptPath );

const defaultProps = {
	prefixClass: 'vector-',
	id: 'searchform',
	searchAccessKey: 'f',
	searchTitle: 'search',
	showThumbnail: true,
	showDescription: true,
	highlightQuery: true,
	urlGenerator,
	restClient: {
		loadMore: () => Promise.resolve(),
		fetchByTitle: () => Promise.resolve()
	},
	searchPlaceholder: 'Search MediaWiki',
	searchQuery: ''
};

const mount = ( /** @type {Object} */ customProps ) => VueTestUtils.shallowMount( App, {
	props: Object.assign( {}, defaultProps, customProps ),
	global: {
		mocks: {
			$i18n: ( /** @type {string} */ str ) => ( {
				text: () => str
			} )
		},
		directives: {
			'i18n-html': ( el, binding ) => {
				el.innerHTML = `${ binding.arg } (${ binding.value })`;
			}
		}
	}
} );

describe( 'App', () => {
	it( 'renders a typeahead search component', () => {
		const wrapper = mount();
		expect(
			wrapper.element
		).toMatchSnapshot();
	} );
} );
