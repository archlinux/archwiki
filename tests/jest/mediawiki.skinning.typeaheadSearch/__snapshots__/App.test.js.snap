// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a typeahead search component 1`] = `
<cdx-typeahead-search-stub
  accesskey="f"
  aria-label="Search MediaWiki"
  autoexpandwidth="false"
  buttonlabel="searchbutton"
  class="vector-typeahead-search"
  debounceinterval="120"
  formaction=""
  highlightquery="true"
  id="searchform"
  initialinputvalue=""
  placeholder="Search MediaWiki"
  search-results-label="searchresults"
  searchfooterurl=""
  searchresults=""
  showthumbnail="true"
  title="search"
  usebutton="false"
  visibleitemlimit="7"
/>
`;
